java-checks/src/main/java/org/sonar/java/checks/MissingDeprecatedCheck.java

```diff
@@ -27,16 +27,15 @@
 import org.sonar.plugins.java.api.tree.AnnotationTree;
 import org.sonar.plugins.java.api.tree.Tree;
 import org.sonar.plugins.java.api.tree.Tree.Kind;
+import org.sonar.plugins.java.api.tree.VariableTree;
 import org.sonarsource.analyzer.commons.annotations.DeprecatedRuleKey;
 
 @DeprecatedRuleKey(ruleKey = "MissingDeprecatedCheck", repositoryKey = "squid")
 @Rule(key = "S1123")
 public class MissingDeprecatedCheck extends AbstractDeprecatedChecker {
 
   private static final Kind[] CLASS_KINDS = PublicApiChecker.classKinds();
```

Comment:
Do you think we should add Records to CLASS_KINDS? It looks like it might affect other checks so is out of the scope for this ticket.

Reply:
As I understand it, when it comes to preview features, we want to do the least possible changes necessary to keep the analyzer from crashing. Records being missing from `CLASS_KINDS` doesn't seem to cause any crashes, so I'd say it should remain unchanged for now.

Reply:
Yep, let's leave it as is.